.github/workflows/integration-tests-vm-type.yml

```diff
@@ -0,0 +1,84 @@
+name: Collector Integration Tests
+
+on:
+  workflow_call:
+    inputs:
+      vm_type:
+        description: "Type of VM to run integration tests on. e.g. rhel or ubuntu-os"
+        type: string
+        required: true
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    env:
+      JOB_ID: ${{ github.run_id }}${{ inputs.vm_type }}
+      GCP_SSH_KEY_FILE: ~/.ssh/GCP_SSH_KEY
+      # temporary hard-code
+      COLLECTOR_TAG: 3.12.0
+      BUILD_TYPE: ci
+      VM_TYPE: "${{ inputs.vm_type }}"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.13' # to match the requirement in the integration tests
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+
+      - name: Install python dependencies
+        run: python -m pip install -r ansible/requirements.txt
+
+      - name: Authenticate with GCP
+        uses: 'google-github-actions/auth@v1'
+        with:
+          credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_COLLECTOR_SVC_ACCT }}'
+
+      - name: Setup GCP
+        uses: 'google-github-actions/setup-gcloud@v1'
+
+      - name: Copy creds files
+        run: |
+          source ${{ github.workspace }}/.openshift-ci/scripts/lib.sh
+
+          mkdir -p /tmp/secret
+          cp "$GOOGLE_APPLICATION_CREDENTIALS" /tmp/secret/gcp-account.json
+
+          mkdir -p "$HOME/.ssh"
+          chmod 0700 "$HOME/.ssh"
+          copy_secret_to_file GCP_SSH_KEY "$HOME/.ssh/GCP_SSH_KEY" 0600
+          copy_secret_to_file GCP_SSH_KEY_PUB "$HOME/.ssh/GCP_SSH_KEY.pub" 0600
+        env:
+          GCP_SSH_KEY: ${{ secrets.GCP_SSH_KEY }}
+          GCP_SSH_KEY_PUB: ${{ secrets.GCP_SSH_KEY_PUB }}
+
+      - name: Normalize job ID
+        run: |
+          NORM_JOB_ID="${JOB_ID/-/}"
+          echo "JOB_ID=${NORM_JOB_ID}" >> "$GITHUB_ENV"
+
+      - name: Create VMs
+        id: create-vms-step
+        run: |
+          make -C "${{ github.workspace }}/ansible" create-vms
+
+      - name: Run Tests
+        run: make -C "${{ github.workspace }}/ansible" integration-tests
+        env:
+          QUAY_RHACS_ENG_RO_USERNAME: ${{ secrets.QUAY_RHACS_ENG_RO_USERNAME }}
+          QUAY_RHACS_ENG_RO_PASSWORD: ${{ secrets.QUAY_RHACS_ENG_RO_PASSWORD }}
+
+      - name: Teardown VMs
+        if: always()
+        run: make -C "${{ github.workspace }}/ansible" destroy-vms
+
+      - name: Store artifacts
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          path: |
+            ${{ github.workspace }}/integration-tests/container-logs/**/*
+            ${{ github.workspace }}/integration-tests/perf.json
+            ${{ github.workspace }}/integration-tests/performance-logs/**/*
```

Comment:
I would also add a name to this artifact, so it doesn't get uploaded as the generic `artifact` name and potentially grouped with other stuff. Maybe something like this?
```yaml
name: ${{ inputs.vm_type }}-logs
```